tests/fields/postgres/test_ranges.py

```diff
@@ -0,0 +1,364 @@
+
+from tests.models import models
+from xocto import localtime, ranges
+
+
+pytestmark = pytest.mark.django_db
+
+
+class TestFiniteDateRangeField:
+    def test_roundtrip(self):
+        finite_date_range = ranges.FiniteDateRange(
+            start=datetime.date(2024, 1, 10), end=datetime.date(2024, 2, 9)
+        )
+        obj = models.FiniteDateRangeModel.objects.create(
+            finite_date_range=finite_date_range
+        )
+        queried = models.FiniteDateRangeModel.objects.get(pk=obj.pk)
+        assert queried.finite_date_range == finite_date_range
+
+    def test_nullable(self):
+        obj = models.FiniteDateRangeModel.objects.create(
+            finite_date_range=ranges.FiniteDateRange(
+                start=datetime.date(2024, 1, 10), end=datetime.date(2024, 2, 9)
+            ),
+            finite_date_range_nullable=None,
+        )
+        queried = models.FiniteDateRangeModel.objects.get(pk=obj.pk)
+        assert queried.finite_date_range_nullable is None
+
+    def test_query(self):
+        finite_date_range = ranges.FiniteDateRange(
+            start=datetime.date(2024, 1, 10), end=datetime.date(2024, 2, 9)
+        )
+        models.FiniteDateRangeModel.objects.create(finite_date_range=finite_date_range)
+        assert models.FiniteDateRangeModel.objects.filter(
+            finite_date_range=finite_date_range
+        ).exists()
+        assert models.FiniteDateRangeModel.objects.filter(
+            finite_date_range__overlap=ranges.FiniteDateRange(
+                start=datetime.date(2024, 1, 1), end=datetime.date(2024, 1, 15)
+            )
+        ).exists()
+        assert models.FiniteDateRangeModel.objects.filter(
+            finite_date_range__contains=ranges.FiniteDateRange(
+                start=datetime.date(2024, 1, 11), end=datetime.date(2024, 1, 15)
+            )
+        ).exists()
+        assert not models.FiniteDateRangeModel.objects.filter(
+            finite_date_range__contains=ranges.FiniteDateRange(
+                start=datetime.date(2024, 1, 5), end=datetime.date(2024, 1, 15)
+            )
+        ).exists()
+
+    def test_query_does_not_allow_tuple_values(self):
+        with pytest.raises(TypeError):
+            models.FiniteDateRangeModel.objects.filter(
+                finite_date_range=(
+                    datetime.date(2024, 1, 10),
+                    datetime.date(2024, 2, 9),
+                )
+            )
+        with pytest.raises(TypeError):
+            models.FiniteDateRangeModel.objects.filter(
+                finite_date_range__overlap=(
+                    datetime.date(2024, 1, 1),
+                    datetime.date(2024, 1, 15),
+                )
+            )
+        with pytest.raises(TypeError):
+            models.FiniteDateRangeModel.objects.filter(
+                finite_date_range__contains=(
+                    datetime.date(2024, 1, 11),
+                    datetime.date(2024, 1, 15),
+                )
+            )
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** This is 3 tests hidden in a trench-coat

There are 3 separate tests here - one for each of the filter lookups. Whilst they are all effectively the same check under the hood, tests should ideally only test 1 thing to ensure that they are specific and can more easily pinpoint an issue to the specific area effected.

Reply:
Fair, but we don't expect these to ever fail, as they all go via the same codepath. It's more a demonstration of that.